First Task/src/test/java/task/FunctionalTest.java

```diff
@@ -0,0 +1,71 @@
+package task;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static java.util.Arrays.asList;
+import static task.Functional.max;
+import static task.Functional.sum;
+
+class FunctionalTest {
+
+    @Test
+    void zeroSum() {
+        assertEquals(0, sum(Collections.emptyList()));
+    }
+
+    @Test
+    void sumOfEqualElements() {
+        assertEquals(5 * 3, sum(asList(5, 5, 5)));
+    }
+
+    @Test
+    void commutativity() {
+        int sum1 = sum(asList(1, 2, 3));
+        int sum2 = sum(asList(3, 2, 1));
+        assertEquals(sum1, sum2);
+    }
+
+    @Test
+    void associativity() {
+        int sum1 = sum(asList(1, 2, 3, 4));
+        int sum2 = sum(asList(1, 2)) + sum(asList(3, 4));
+        assertEquals(sum1, sum2);
+    }
+
+    @Test
+    void maxInEmptyList() throws NoSuchElementException{
```

Comment:
There are two types of exceptions, they are `Checked` and `Unchecked`.

`Checked` exceptions are exceptions you have to either handle or throw further. For instance, `IOException` is `Checked`, your code won't compile if it uses some code that throws `IOException` and does not handle it throw further.

`Unchecked` exceptions are exceptions that are not necessary to handle at all and define in the method signature. All the subclasses of `RuntimeException` are `Unchecked` you don't need to write them in the method definition or catch, cause it is not always possible.

As you can check by yourself `NoSuchElementException` is `Unchecked` since it extends `RuntimeException`.